examples/brute_force/brute_force.cpp

```diff
@@ -0,0 +1,166 @@
+  static KOKKOS_FUNCTION auto get(Dummy const &, size_type i)
+  {
+    return attach(
+        intersects(Sphere{{{(float)i, (float)i, (float)i}}, (float)i}), i);
+  }
+};
+
+struct PrintfCallback
+{
+  Kokkos::View<int, ExecutionSpace, Kokkos::MemoryTraits<Kokkos::Atomic>> c_;
+  PrintfCallback()
+      : c_{"counter"}
+  {
+  }
+  template <typename Predicate, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate const &predicate, int i,
+                                  OutputFunctor const &out) const
+  {
+    int const j = getData(predicate);
+    printf("%d callback (%d,%d)\n", ++c_(), i, j);
+    out(i);
+  }
+};
+
+template <typename T, typename... P>
+std::vector<T> view2vec(Kokkos::View<T *, P...> view)
+{
+  std::vector<T> vec(view.size());
+  Kokkos::deep_copy(Kokkos::View<T *, Kokkos::HostSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+                        vec.data(), vec.size()),
+                    view);
+  return vec;
+}
+
+template <typename OutputView, typename OffsetView>
+void print(OutputView const out, OffsetView const offset)
+{
+  int const n_queries = offset.extent(0) - 1;
+
+  auto const h_out = view2vec(out);
+  auto const h_offset = view2vec(offset);
+  int count = 0;
+
+  for (int j = 0; j < n_queries; ++j)
+    for (int k = h_offset[j]; k < h_offset[j + 1]; ++k)
+      printf("%d result (%d, %d)\n", ++count, h_out[k], j);
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  int nqueries = 5, nprimitives = 5, nrepeats = 1;
+  namespace bpo = boost::program_options;
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "predicates", bpo::value<int>(&nqueries), "number of predicates" )
+      ( "primitives", bpo::value<int>(&nprimitives), "number of primitives" )
+      ( "iterations", bpo::value<int>(&nrepeats), "number of iterations" )
+      ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  if (vm.count("help") > 0)
+  {
+    std::cout << desc << '\n';
+    return 1;
+  }
+  printf("Primitives: %d\n", nprimitives);
+  printf("Predicates: %d\n", nqueries);
+  printf("Iterations: %d\n", nrepeats);
+
+  ExecutionSpace space{};
+  Dummy primitives{nprimitives};
+  Dummy predicates{nqueries};
+
+  for (int i = 0; i < nrepeats; i++)
+  {
+    int out_count;
+    {
+      Kokkos::Timer timer;
+      ArborX::BoundingVolumeHierarchy<MemorySpace> bvh{space, primitives};
+
+      Kokkos::View<int *, ExecutionSpace> indices("indices_ref", 0);
+      Kokkos::View<int *, ExecutionSpace> offset("offset_ref", 0);
+      bvh.query(
+          space, predicates, ArborX::Details::DefaultCallback{}, indices,
+          offset,
+          ArborX::Experimental::TraversalPolicy{}.setPredicateSorting(true));
+
+      double time = timer.seconds();
```

Comment:
I think we need to add a `space.barrier()` before this when using with Cuda. One of the potential issues is that `Kokkos::parallel_for` (that is being used inside `query`) may return immediately before the execution of the kernel finishes. This may affect this timing result. I've seen this happen with a pure callback, and not 100% sure it will happen with this form, but we should be conservative here.